crypto/x509/v3_skid.c

```diff
@@ -52,55 +52,48 @@ ASN1_OCTET_STRING *s2i_ASN1_OCTET_STRING(X509V3_EXT_METHOD *method,
 
 }
 
-static ASN1_OCTET_STRING *s2i_skey_id(X509V3_EXT_METHOD *method,
-                                      X509V3_CTX *ctx, char *str)
+ASN1_OCTET_STRING *X509_PUBKEY_hash(X509_PUBKEY *pubkey)
 {
     ASN1_OCTET_STRING *oct;
-    X509_PUBKEY *pubkey;
     const unsigned char *pk;
     int pklen;
     unsigned char pkey_dig[EVP_MAX_MD_SIZE];
     unsigned int diglen;
 
-    if (strcmp(str, "hash"))
-        return s2i_ASN1_OCTET_STRING(method, ctx, str);
-
-    if ((oct = ASN1_OCTET_STRING_new()) == NULL) {
-        ERR_raise(ERR_LIB_X509V3, ERR_R_MALLOC_FAILURE);
+    if (pubkey == NULL) {
+        ERR_raise(ERR_LIB_X509V3, X509V3_R_NO_PUBLIC_KEY);
         return NULL;
     }
+    if ((oct = ASN1_OCTET_STRING_new()) == NULL)
+        return NULL;
 
-    if (ctx && (ctx->flags == CTX_TEST))
+    X509_PUBKEY_get0_param(NULL, &pk, &pklen, NULL, pubkey);
+    if (EVP_Digest(pk, pklen, pkey_dig, &diglen, EVP_sha1(), NULL)
```

Comment:
Please add TODO(3.0) - explicitly fetch the digest.

Reply:
Done